code/feature_extraction/Util.py

```diff
@@ -0,0 +1,95 @@
+class Util:
+    """
+    CharCounter:
+
+        ASSUMPTIONS:
+            - Special characters do count for the `character length' but not as a word
+            - The `diff' methods return the absolute value
+
+        OBSERVATIONS:
+            - postTitle does not exist, just postText --> "[<text of the post with links removed>]". Analysis showed that the postText contains at most 1 element (so Y U an array?)
+
+        FEATURE IDEAS:
+            - Post Title same as Article Title (0,1)
+                Maybe preprocess retweets e.g. [RT @fionamatthias: 10 ways the expat life Is like a continual espresso buzz via @WSJ] -> 10 Ways the Expat Life Is Like a Continual Espresso Buzz - Expat
+            - Match known clickbait indicators
+                https://www.quora.com/What-are-the-most-commonly-used-words-in-clickbait-titles-Those-news-headlines-that-go-like-You-wont-believe-___-or-This-___-will-restore-your-faith-in-humanity-X-easy-tricks-to-___-They-hint-at-the-article-and-tempt-you-to-click
+                https://www.cnet.com/news/facebook-nixes-click-bait-headlines-in-users-news-feeds/
+
+    WordCounter:
+        WORD COUNT -- Harder Than It Sounds
+            Original approach: len(sentence.split()), but this also counts punctuation (e.g. '--' or '&') as words
+            Second approach: regex  re.compile(r'\S+')  (failed to recognize decimal numbers)
+            Third approach: cleaning text and split on space, problem: string.punctuation too broad
+              --> (sentence.translate(str.maketrans(' ', ' ', string.punctuation)))
+
+        So, problems:
+            word--word     (are two words)
+            U.S.           (is one word)
+            2.5            (is one 'word')
+            missing'space' (are two words)
+    """
+
+    @staticmethod
+    def ratio(left, right):
+
+        # Catch edge cases
+        # TODO: check if `<= 0' check makes sense
+        if not left or left <= 0 or not right or right <= 0:
```

Comment:
I think that if left == 0, the ratio is still defined as `ratio = 0 / right = 0`

Reply:
True, but I'm a little confused by the "exists" operator here:
![image](https://user-images.githubusercontent.com/579633/55151733-2cf48780-514f-11e9-883d-5fe35ef291db.png)

This would mean that both `left` and `right` should exist, but `right` should also not be zero, so that's why I though that (by this definition) left also shouldn't be zero. Even more so because they state:
> "We calculate only one side of the equation. The other side is the opposite of the first."
![image](https://user-images.githubusercontent.com/579633/55151975-b3a96480-514f-11e9-88cc-d1e5ce3c6748.png)

The lower than or equal is because when we get a `-1` here, it means that it "doesn't exist" so following the definition the "ratio" should also not exist.

My 2 cents ;p

Reply:
agreed